featurebyte/query_graph/node/metadata/operation.py

```diff
@@ -177,29 +177,30 @@ class BaseDerivedColumn(BaseColumn):
 
     @staticmethod
     def insert_column(
-        column_map: Dict[str, BaseDataColumn], column: BaseDataColumn
-    ) -> Dict[str, BaseDataColumn]:
+        column_map: Dict[Tuple[str, str], BaseDataColumn], column: BaseDataColumn
+    ) -> Dict[Tuple[str, str], BaseDataColumn]:
         """
         Insert column into dictionary. If more than more columns with the same name, aggregate the node names
         (make sure we don't miss any node which is used for pruning) and combine filter flag (take the max
         value to indicate the filtered column has been used at least in one operation).
 
         Parameters
         ----------
-        column_map: Dict[str, BaseDataColumn]
+        column_map: Dict[Tuple[str, str], BaseDataColumn]
             Dictionary map
         column: BaseDataColumn
             Data column object
 
         Returns
         -------
-        Dict[str, BaseDataColumn]
+        Dict[Tuple[str, str], BaseDataColumn]
         """
```

Comment:
Update the logic here so that it won't combine the column with the same name but from different table.